src/Sniffs/CamelCaseParameterSniff.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ShopSys\CodingStandards\Sniffs;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\AbstractVariableSniff;
+use PHP_CodeSniffer\Util\Common;
+
+final class CamelCaseParameterSniff extends AbstractVariableSniff
+{
+    /**
+     * @var string
+     */
+    private const NOT_LETTER_THEN_LETTER_PATTERN = '([^a-zA-Z]{1}(?<nextLetter>[A-Za-z]{1}))';
+
+    /**
+     * @param int $position
+     */
+    protected function processVariable(File $file, $position): void
+    {
+        $currentToken = $file->getTokens()[$position];
+
+        $variableName = ltrim($currentToken['content'], '$');
+        if (Common::isCamelCaps($variableName)) {
+            return;
+        }
+
+        $fix = $file->addFixableError(sprintf(
+            'Variable "$%s" should be camel case',
+            $variableName
+        ), $position, self::class);
+
+        if ($fix) {
+            $this->fixVariableName($file, $position, $variableName);
+        }
+    }
+
+    /**
+     * @param int $position
+     */
+    protected function processVariableInString(File $file, $position): void
+    {
+    }
+
+    /**
+     * @param int $position
+     */
+    protected function processMemberVar(File $file, $position): void
+    {
+    }
+
+    /**
+     * Changes:
+     * _someVariable => someVariable
+     * some_variable => someVariable
+     */
+    private function fixVariableName(File $file, int $position, string $variableName): void
+    {
+        $newVariableName = preg_replace_callback(
+            '#(?<prefixLetter>[a-z]{1})?' . self::NOT_LETTER_THEN_LETTER_PATTERN . '#',
+            function (array $match): string {
```

Comment:
Let me know if comment + method is clear in what it does. 
I'll improve the naming and add more examples if not

Reply:
We discussed this with @PetrHeinz and he was not satisfied with the solution (eg. we came up with few use cases when this solution is not sufficient and it does not take numbers into account). We suggest to copypaste the `PHP_CodeSniffer\Standards\Squiz\Sniffs\NamingConventions\ValidVariableNameSniff` into new `ShopSys\CodingStandards\Sniffs\ValidVariableNameSniff` and just remove the logic of ignoring private fields prefixed with underscore. What do you think?

Reply:
I see. Could you share these use cases? That's the most important information here.

`PHP_CodeSniffer\Standards\Squiz\Sniffs\NamingConventions\ValidVariableNameSniff` is already used actually. **This sniff takes care about `_` prefixes  in private properties.** Together, they should cover it all. If not, it's a bug that I shall fix.

Copy-pasting whole sniff in from 3rd party is fast idea, that might fix it the issue today.  But eventually could lead to:

- missing update
- missing tests
- lot of code that is not understandable and difficult to maintain and develop, which PHP_CodeSniffer code sure is

We could use original Sniff thought, that's a good idea. There is advanced  regular that checks this: https://github.com/squizlabs/PHP_CodeSniffer/blob/818f188f71940f9cf848eca29b037cbaa2cde2f2/src/Standards/Squiz/Sniffs/NamingConventions/ValidVariableNameSniff.php#L188

But, if the reported cases are nearly impossible to fix, we might give it a try :+1:

Reply:
@vitek-rostislav Btw, this is why story is so important. 

- *It doesn't work => solution* is like "git squash to 1 commit"
- *It dones't work on these cases => this is probably the best solution* is like having the full commit hiStory :+1:

Reply:
Ok, fair enough, you are right that copypasting 3rd party code is not a good solution.

One of the problems is that there is used method `Common::isCamelCaps($variableName)` to determine whether the variable is in camelCase and than there is used different logic for fixing the variable which does not take into account numbers, PHP superglobals (even though they are covered by different sniff), unicode symbols, and edge cases like `$$var` or having `$aB` and `$a_b` in the same file, where fixing of such thing could break the application. Maybe the main problem here is that we do not have specified all use cases that we want to cover with the fixer. So I suggest not to implement the fix at the time, and we can do it afterwards we define the use cases properly.

Unfortunately, I also found another problem with the sniff itself, it does not report declaration of private variable in underscore as a violation.

Reply:
> So I suggest not to implement the fix at the time, and we can do it afterwards we define the use cases properly.

All right, that makes sense. I'll remove it for now then

Reply:
Done :+1: